skyportal/handlers/api/internal/recent_sources.py

```diff
@@ -89,5 +89,12 @@ def get(self):
             # Delete bookkeeping recency_index key
             del source["recency_index"]
 
-        self.verify_and_commit()
+        try:
+            # check for AccessError:
+            # this is in a try-except in case of group deletions
+            # which lead to a change in what classifications are accessible
+            self.verify_and_commit()
+        except AccessError as e:
+            return self.error(f'Classification access failed: {e}')
```

Comment:
I dont think it makes sense to show classification access errors to the end users since they did not explicitly ask for classifications. Perhaps just log the error using `log()`?

I'm a bit concerned here that we may not have the cascade of classification permissions correct. A user should be able to see classifications that they have made (which means classifications should automatically belong to that user's solo group) and not classifications saved to groups in which they do not belong. I would have thought that the `source_entry = (Source.query_records_accessible_by(self.current_user)...` call would have been enough to enforce correct classifications viewership.

Reply:
I think Stefan and I resolved this yesterday. We needed a source refresh.